DoSomething/gladiator
# Mail queue

#### What's this PR do?

* Added the `QueueMessageRequest` event and `QueueMessage` event listener that fire and listen when we need to send and email. The listener implements the `ShouldQueue` Interface that pushes the job onto the redis queue. 

* Adds a `MessagesController` that kicks of the event to send the message when called. 

* A very basic reminder email view for testing...so we get some content in there

* Updates to `.env.example` that show the new settings that we need to run the queue and mandril driver. 

#### How should this be manually tested?
Pull down branch. 
Update `.env` with correct settings. (mandril key is in mandrill, login with lastpass)
Set up a message in your DB

Run queue listener:
`php artisan queue:listen`

Run command in `art tinker`
`Gladiator\Http\Controllers\MessagesController::sendMessage($message_id)`

You should get an email...or actually, I, will get an email cause my email address is hardcoded right now.

#### Any background context you want to provide?

This just does the basic set up work of sending emails in Gladiator, we will still have to add logic to send to actual people and need to wait for @weerd 's work to be finised on #137 

I also mentioned on slack that we are gonna have to boot up the [queue listener on gladiator deploys](https://github.com/DoSomething/gladiator/blob/master/config/deploy.rb#L46-L48) similar to what is done in [longshot](https://github.com/DoSomething/longshot/blob/5a20eb9383d2f11896bfd48998a4c8f3e61ac4f9/config/deploy.rb#L55)

Also noting that we should do some logging for succesful email sends and failures.

#### What are the relevant tickets?
Fixes #132 

@angaither @weerd @deadlybutter
